### ISSUE TITLE
Regression from csc (C# 5): Type <type> is defined in an assembly that is not referenced.

### ISSUE THREAD

**Issue by Imisnew2** [dotnet/roslyn#15588]
**Version Used**:
I used the following versions:
* .NET Core 1.1
* .NET Core Tools MSBuild "alpha"

Note that the below steps are using the .NET Core Tools MSBuild "alpha" version, although you can tweak them easily for any .NET Core version.

**Steps to Reproduce**:
1. Create the following source file:
`
using Microsoft.DirectX.Direct3D;
class BadCS0012
{
    void Test(Device d)
    {
        d.VertexFormat = CustomVertex.TransformedColoredTextured.Format;
    }
}
`
2. Place the attached assemblies into the same directory as the source file [(Direct3D-and-CoreLib-dlls.zip)](https://github.com/dotnet/roslyn/files/620285/Direct3D-and-CoreLib-dlls.zip):
* Microsoft.DirectX.Direct3D.dll
* System.Private.CoreLib.dll
3. Compile the source file:
* (C# 5): Using 'C:/Windows/Microsoft.NET/Framework64/v4.0.30319/csc.exe', run: `csc /noconfig /t:library /r:Microsoft.DirectX.Direct3D.dll BadCS0012.cs`
* (Roslyn): Using 'C:/Program Files/dotnet/dotnet.exe' and 'C:/Program Files/dotnet/sdk/1.0.0-preview3-004056/csc.exe', run: `dotnet csc /noconfig /t:library /r:Microsoft.DirectX.Direct3D.dll /r:System.Private.CoreLib.dll BadCS0012.cs`

**Expected Behavior**:
(C# 5 csc) exhibits the same behavior as (Roslyn csc), or has a known/good reason not to.

**Actual Behavior**:
(C# 5 csc) successfully compiles and (Roslyn csc) fails with CS0012: The type 'IsConstModifier' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.VisualC, Version=7.0.5000.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3'.

**Comment by gafter** [comment]
Likely dup of #9370

**Comment by Imisnew2** [comment]
@gafter 
FWIW, there are no methods here to perform overload resolution with (unless you consider `set_VertexFormat(...)`):
* CustomVertex is a `sealed class`.
* TransformedColoredTextured is an inner `struct`.
* Format is a `const VertexFormats` field.
* Finally, VertexFormats is an `enum`.

Perhaps there's something I'm not immediately noticing w.r.t. member / overload resolution which causes this error... Note that 'IsConstModifier' is never referenced (I think?). Looking at nearby methods, it appears to be used inside of method bodies with 'calli' instructions via use of 'modopt(Microsoft.VisualC.IsConstModifier)'.

**Comment by gafter** [comment]
@Imisnew2 Yes, I see that. Nevertheless, I suspect the root cause may be the same, and may have nothing to do with overload resolution.
